### PR TITLE
Staff and instructor users of the master course are added to CCX created via REST APIs

### LINKED ISSUE #205: Add staff and instructors users as staff of a CCX created via REST APIs
comment https://github.com/edx/edx-platform/pull/10877#issuecomment-173222178

### PR DESCRIPTION
**Pull request by giocalitri**
#### What are the relevant tickets?
fixes https://github.com/mitocw/edx-platform/issues/205

#### What's this PR do?
Adds the master course staff users to a new CCX created via the CCX REST APIs 

#### Where should the reviewer start?
The added functionality is in `lms.djangoapps.ccx.api.v0.views`, but there has been a significant change of code in `lms.djangoapps.ccx.utils`.

#### How should this be manually tested?
A staff user on a master course should be able to view and access a CCX created via REST API using the normal dashboard.

#### Any background context you want to provide?
The added functionality of this PR is the equivalent of what https://github.com/edx/edx-platform/pull/10877 did with `lms.djangoapps.ccx.views`.

### CODE REVIEW

File: lms/djangoapps/ccx/api/v0/tests/test_views.py [hunk 1]
```diff
@@ -459,6 +462,38 @@ def test_post_list(self):
...
+        course_key = CourseKey.from_string(resp.data.get('ccx_course_id'))  # pylint: disable=no-member
+        with ccx_course_cm(course_key) as course_ccx:
+            list_staff_ccx_course = list_with_level(course_ccx, 'staff')
+            list_instructor_ccx_course = list_with_level(course_ccx, 'instructor')
+
+        self.assertEqual(len(list_staff_master_course), len(list_staff_ccx_course))
+        for course_user, ccx_user in izip(sorted(list_staff_master_course), sorted(list_staff_ccx_course)):
+            self.assertEqual(course_user, ccx_user)
+        self.assertEqual(len(list_instructor_master_course), len(list_instructor_ccx_course))
+        for course_user, ccx_user in izip(sorted(list_staff_master_course), sorted(list_staff_ccx_course)):
```
Comment (noisecapella) [inline lms/djangoapps/ccx/api/v0/tests/test_views.py]: It looks like you already did this comparison in the previous for loop

**Comment by pdpinch** [comment]
Can we mention in the title that this is for the REST API?

**Comment by giocalitri** [comment]
@pdpinch: done

File: lms/djangoapps/ccx/tests/test_utils.py [hunk 2]
```diff
@@ -108,3 +114,69 @@ def test_add_master_course_staff_to_ccx(self):
...
+        list_instructor_master_course = list_with_level(self.course, 'instructor')
+        self.assertEqual(len(outbox), 0)
+
+        # run the assignment the first time
+        add_master_course_staff_to_ccx(self.course, self.ccx_locator, self.ccx.display_name)
+        self.assertEqual(len(outbox), len(list_staff_master_course) + len(list_instructor_master_course))
+        with ccx_course(self.ccx_locator) as course_ccx:
+            list_staff_ccx_course = list_with_level(course_ccx, 'staff')
+            list_instructor_ccx_course = list_with_level(course_ccx, 'instructor')
+        self.assertEqual(len(list_staff_master_course), len(list_staff_ccx_course))
```
Comment (noisecapella) [inline lms/djangoapps/ccx/tests/test_utils.py]: Is it possible to compare the lists directly instead of the count, or is that cumbersome?

Comment (noisecapella) [inline lms/djangoapps/ccx/tests/test_utils.py]: Actually it looks like you could just say

    self.assertEqual(sorted(list_staff_master_course), sorted(list_staff_ccx_course))

And you could remove the next couple lines as well

**Comment by noisecapella** [comment]
Looks good, feel free to ignore my comment about the test if it's complicated to do :+1:

File: lms/djangoapps/ccx/tests/test_utils.py (hunk 2 above)
Comment (giocalitri) [inline lms/djangoapps/ccx/tests/test_utils.py]: I actually tried and for some reason it was failing

File: lms/djangoapps/ccx/tests/test_utils.py [hunk 3]
```diff
@@ -62,7 +67,7 @@ def test_ccx_locator(self):
         self.assertEqual(result, ccx)
 
 
-class TestStaffOnCCX(CcxTestCase, SharedModuleStoreTestCase):
```
Comment (pdpinch) [inline lms/djangoapps/ccx/tests/test_utils.py]: Why did you remove SharedModuleStoreTestCase ? Doesn't that make the tests faster?

Comment (giocalitri) [inline lms/djangoapps/ccx/tests/test_utils.py]: because `CcxTestCase` already subclasses it:
https://github.com/mitocw/edx-platform/blob/master/lms/djangoapps/ccx/tests/utils.py#L26

**Comment by noisecapella** [comment]
Was this merged in edX already?

**Comment by giocalitri** [comment]
closing this because included in some work @amir-qayyum-khan already merged in edx
